crates/db_views_moderator/src/modlog_combined_view.rs

```diff
@@ -0,0 +1,1405 @@
+          query.filter(modlog_combined::mod_remove_post_id.is_not_null())
+        }
+        ModlogActionType::ModLockPost => {
+          query.filter(modlog_combined::mod_lock_post_id.is_not_null())
+        }
+        ModlogActionType::ModFeaturePost => {
+          query.filter(modlog_combined::mod_feature_post_id.is_not_null())
+        }
+        ModlogActionType::ModRemoveComment => {
+          query.filter(modlog_combined::mod_remove_comment_id.is_not_null())
+        }
+        ModlogActionType::ModRemoveCommunity => {
+          query.filter(modlog_combined::mod_remove_community_id.is_not_null())
+        }
+        ModlogActionType::ModBanFromCommunity => {
+          query.filter(modlog_combined::mod_ban_from_community_id.is_not_null())
+        }
+        ModlogActionType::ModAddCommunity => {
+          query.filter(modlog_combined::mod_add_community_id.is_not_null())
+        }
+        ModlogActionType::ModTransferCommunity => {
+          query.filter(modlog_combined::mod_transfer_community_id.is_not_null())
+        }
+        ModlogActionType::ModAdd => query.filter(modlog_combined::mod_add_id.is_not_null()),
+        ModlogActionType::ModBan => query.filter(modlog_combined::mod_ban_id.is_not_null()),
+        ModlogActionType::ModHideCommunity => {
+          query.filter(modlog_combined::mod_hide_community_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgePerson => {
+          query.filter(modlog_combined::admin_purge_person_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgeCommunity => {
+          query.filter(modlog_combined::admin_purge_community_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgePost => {
+          query.filter(modlog_combined::admin_purge_post_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgeComment => {
+          query.filter(modlog_combined::admin_purge_comment_id.is_not_null())
+        }
+        ModlogActionType::AdminBlockInstance => {
+          query.filter(modlog_combined::admin_block_instance_id.is_not_null())
+        }
+        ModlogActionType::AdminAllowInstance => {
+          query.filter(modlog_combined::admin_allow_instance_id.is_not_null())
+        }
+      }
+    }
+
+    let mut query = PaginatedQueryBuilder::new(query);
+
+    let page_after = self.page_after.map(|c| c.0);
+
+    if self.page_back.unwrap_or_default() {
+      query = query.before(page_after).limit_and_offset_from_end();
+    } else {
+      query = query.after(page_after);
+    }
+
+    // Tie breaker
+    query = query.then_desc(key::published).then_desc(key::id);
+
+    let res = query.load::<ModlogCombinedViewInternal>(conn).await?;
+
+    // Map the query results to the enum
+    let out = res.into_iter().filter_map(|u| u.map_to_enum()).collect();
+
+    Ok(out)
+  }
+}
+
+impl InternalToCombinedView for ModlogCombinedViewInternal {
+  type CombinedView = ModlogCombinedView;
+
+  fn map_to_enum(&self) -> Option<Self::CombinedView> {
```

Comment:
```suggestion
  fn map_to_enum(self) -> Option<Self::CombinedView> {
```

Then theres no need for any clone below.